server/src/test/java/org/elasticsearch/cluster/coordination/CoordinationDiagnosticsServiceTests.java

```diff
@@ -923,15 +923,16 @@ public void testBeginPollingClusterFormationInfo() {
                 .toList();
             cluster.clusterNodes.stream().filter(node -> node.getLocalNode().isMasterNode()).forEach(node -> {
                 ConcurrentMap<DiscoveryNode, ClusterFormationStateOrException> nodeToClusterFormationStateMap = new ConcurrentHashMap<>();
+                node.coordinationDiagnosticsService.clusterFormationInfoTasks = new ConcurrentHashMap<>();
+                node.coordinationDiagnosticsService.disableAutoPollingForTestMode = true;
                 node.coordinationDiagnosticsService.beginPollingClusterFormationInfo(
                     masterNodes,
                     nodeToClusterFormationStateMap::put,
-                    cancellable -> {}
+                    node.coordinationDiagnosticsService.clusterFormationInfoTasks
                 );
 
                 cluster.runRandomly(false, true, EXTREME_DELAY_VARIABILITY);
                 cluster.stabilise();
-
                 /*
                  * The cluster has now run normally for some period of time, so check that the outputs of
                  * beginPollingClusterFormationInfo() are present with no exceptions:
```

Comment:
I think this is not true anymore and we shouldn't rely on this condition - the polling was cancelled at some point after the cluster was stabilised